### ISSUE TITLE
Heat speed loss should depend on time spent in heat rather than temperature

### ISSUE THREAD

**Issue by echomirage1** [CleverRaven/Cataclysm-DDA#29711]
**Is your feature request related to a problem? Please describe.**  
Dealing with fire decreases speed too abruptly and this creates silly situations and sudden deaths. I guess this is to simulate heatstroke, which depends more on the amount of time spent in the heat, rather than the intensity of it. Currently speed loss from pain and heat together creates instant death situations.

Death 1: Midgame survivor with decent armor. Saw magma rift, tried to loot sulfur. When I try to grab the second chunk of sulfur, a corrosive zombie appears from behind the nearby house and machineguns me down while I try to move 3-5 tiles.

Death 2: Try to jackhammer out a lab library door, burning portal 2 rooms away starts scorching me. I stop after 5 minutes of hammering and try to run towards a nearby staircase. Peek down, room full of secubots. They follow me back through the staircase and gun me down while I try to run away.

Death 3: Lab start, try to throw molotov at the turret at the exit. The molotov lands at my feet. I burn to death without being able to move one tile away from the fire.

**Describe the solution you'd like**  
Speed loss from heat should be handled with a status effect that depends more on time spent in heat rather than the temperature, simulating the effects of heatstroke, similarly to frostbite. Pain from fire already debuffs speed.

**Additional context**  
People on fire usually are the fastest men alive. People prone to panic when on fire, but they are not sluggish.

Edit: Deleted complaints about lack of stop, drop and roll.

**Comment by Phenomphear** [comment]
If you stand still and press numpad 5 to pass a turn or w/e your keybind is for it you'll roll on the ground to put out the fire. There is no animation for this only a log message and there is only a brief tip on the menu about what to do when you're on fire so I understand your frustration.

I feel the same way about heat sources and their immediate impact on the player. I've gone from full and slaked to dehydrated and famished after cooking 2 items in a fire from a brazier inside the shelter.

It seems fire's effect on the player needs a look over and maybe bring some of the effects down a little.

**Comment by echomirage1** [comment]
@Phenomphear I didn't know you could do that. Thanks for the tip, I'll edit the post.

**Comment by Phenomphear** [comment]
You're welcome. oddly enough you can do this in a moving vehicle if you "let go of the controls".... o.O.

Found that out when I was trying to ram my way into a national guard camp and got set on fire by some enemy.

**Comment by kevingranade** [comment]
If you're interested in contributing, please locate some sources that outline what the timelines are for extreme heat exposure. "Base it more on time" doesn't give us anything to go on.

**Comment by Phenomphear** [comment]
Sure thing, I'll start up debug mode and play around in the shelter with some different fire sources

**Comment by Phenomphear** [comment]
![image](https://user-images.githubusercontent.com/28910998/56459507-7a78a480-6349-11e9-8c2f-f5a84e8ba8fd.png)

I don't know how to spawn a burning portal, I got a bunch of shimmering ones. The only thing I noticed is that if you really close to a huge fire, yea it can be bad for you. I do remember in the past with certain conditions while you were cooking, if you put too much fuel in your fire the room could get really hot really fast and would hurt you.
